packages/sources/coingecko/src/endpoint/global.ts

```diff
@@ -11,16 +11,23 @@ const customError = (data: any) => {
 const customParams = {
   market: ['quote', 'to', 'market', 'coin'],
   path: true,
+  endpoint:false
 }
 
 export const execute: ExecuteWithConfig<Config> = async (request, config) => {
   const validator = new Validator(request, customParams)
   if (validator.error) throw validator.error
 
-  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || config.DEFAULT_ENDPOINT
+  if (endpoint.toLowerCase() === 'globalmarketcap')
```

Comment:
Hmm.. this is an extra piece that we should have a pattern for. We want `execute` to be agnostic of Chainlink endpoints.

So we should handle it in `buildSelector` by holding a mapping of what endpoint -> path should be used here.
```
const endpointPaths = {
globalmarketcap: 'total_market_cap',
dominance: 'market_cap_percentage'
}
```

There are scenarios where the path needs to be dynamically created based off of an input parameter, so there should be an ability to use a function.

We should pair up on this.

Reply:
Here is an example of where this is kind of already happening:
https://github.com/smartcontractkit/external-adapters-js/pull/662/files#diff-d34e11b22b818ada3cee974f527fb51fb6416f6d49b4eb753790a6c6483cbcc8R7

But we need more robust patterns around it and the ability to handle it in `buildSelector`.